src/view/com/posts/Feed.tsx

```diff
@@ -51,16 +64,23 @@ export const Feed = observer(function Feed({
   headerOffset?: number
   ListHeaderComponent?: () => JSX.Element
   extraData?: any
+  showFeedHeaderContextMenu?: boolean
 }) {
   const pal = usePalette('default')
   const theme = useTheme()
   const {track} = useAnalytics()
   const [isRefreshing, setIsRefreshing] = React.useState(false)
+  const params = feed.params as AppBskyFeedGetFeed.QueryParams
+  const isCustomFeed = Boolean(params.feed)
 
   const data = React.useMemo(() => {
     let feedItems: any[] = []
     if (feed.hasLoaded) {
-      if (feed.hasError) {
+      if (isCustomFeed) {
+        feedItems.push(FEED_INNER_HEADER)
+      }
+      if (feed.hasError && !isCustomFeed) {
+        // applies to our internal algo feeds only
```

Comment:
If it's a custom feed, the error message will show within the "inner" header component, which this PR introduces. If it's not a custom feed, we show a similar looking error message outside the header as we have been.